src/main/java/hudson/tasks/test/AbstractTestResultAction.java

```diff
@@ -80,6 +88,27 @@
 
     private Map<String,String> descriptions = new ConcurrentHashMap<String, String>();
 
+    /**
+     * Tool tips for "Overall Build Analysis" trend type.
+     */
+    private Map<NumberOnlyBuildLabel,String> failToolTip, totalToolTip, skipToolTip;
```

Comment:
Why are we not initializing the `Map` here ? Also use `ConcurrentHashMap` instead of `HashMap`

Reply:
I only initialize the map if its null, as if its not the its contents can be reused. And, I was also thinking of making these maps "transient" to avoid serializing them. And, I will replace the ConcurrentHashMap with HashMap, I made it HashMap because it didn't seem like multiple threads will be accessing these maps concurrently but I will make the changes.